solidity/contracts/Greeter.sol

```diff
@@ -38,9 +40,14 @@ contract Greeter is IGreeter {
     }
 
     greeting = _greeting;
+    _updateLastGreetingSetTime(block.timestamp);
     emit GreetingSet(_greeting);
   }
 
+  function _updateLastGreetingSetTime(uint256 _timestamp) internal virtual {
+    lastGreetingSetTime = _timestamp;
+  }
```

Comment:
hmm there's no unit test for this internal, wdyt if we just add an internal virtual empty `_beforeGreeting(){}` hook, do that we don't add complexity to the code?

Reply:
agreed on keeping this method to test the data sent (_timestamp)